torch/csrc/jit/codegen/cuda/lower_loops.cpp

```diff
@@ -88,137 +103,83 @@ void LoopNestGenerator::handle(const Expr* expr) {
           " cannot lower ",
           out->getValType().value());
 
-      pushBack(ir_builder_.create<kir::Allocate>(
+      pushFront(ir_builder.create<kir::Allocate>(
           gpu_lower->lowerValue(out),
           MemoryType::Local,
-          ir_builder_.create<kir::Int>(1)));
+          ir_builder.create<kir::Int>(1)));
     }
-    pushBack(gpu_lower->lowerExpr(expr));
     return;
   }
 
-  TensorView* out = expr->output(0)->as<TensorView>();
+  TensorView* out_tv = expr->output(0)->as<TensorView>();
 
   // Figure out what the entire loop structure should look like.
   std::deque<IterDomain*> loop_structure;
 
-  // As we go through iteration domains track the previous view
-  const TensorView* last_ca_view = nullptr;
-  // Check where in the previous view our last axis was in that view
-  int64_t last_ca_view_ind = 0;
-
-  // Look at each axis individually in out's domain
-  for (int64_t out_i = 0; out_i < (int64_t)out->getThisComputeAtAxis();
+  // Look at each axis individually in out's domain, first only setup loop
+  // structure within computeAt
+  for (int64_t out_i = 0; out_i < (int)out_tv->getThisComputeAtAxis();
        out_i++) {
-    // Grab the axis information
-    auto ca_point = out->getComputeAtAxis(out_i);
-    auto ca_view = ca_point.second;
-    auto ca_id = ca_point.first;
-
-    // Figure out if there are axes in the compute at tensor view that aren't
-    // in out, make sure to also open them. Check where to start looking for
-    // them in the compute at view.
-    size_t start = 0;
-    if (last_ca_view == nullptr) {
-      // Start at the begining, we haven't processed any axes yet.
-      start = 0;
-    } else if (last_ca_view == ca_view) {
-      // This view is the same as the last axis, so start where we left off.
-      start = last_ca_view_ind + 1;
-    } else {
-      // This is a new view, figure out where we are in it, and start from there
-      for (start = 0; start < ca_view->nDims(); start++) {
-        if (loop_structure.back() == ca_view->getComputeAtAxis(start).first) {
-          break;
-        }
-      }
-      start++;
-    }
+    // Safe to use loop map since this is outside the compute at point
+    auto concrete_id =
+        gpu_lower->caParallelMap().getConcreteMappedID(out_tv->axis(out_i));
```

Comment:
Shouldn't this be `caLoopMap()` instead of `caParallelMap()`?

Reply:
No, you're right, it should be loop because of the misaligned bcast behavior.

Reply:
Hmm, I tried `caLoopMap()` and got a validation error with `FusionPersistentNormLocalShared`. Will look into it.

Reply:
Parallel map should be fine for now. Although in the future we'll want to move to generating a reference either before or during the loop construction and including the reference iter domains in the iter domain maps. The issue that could happen from here is if one of the TensorViews doesn't resolve all the broadcasts that are resolved in a loop nest. This doesn't happen in our current testing but could be done.